assets/js/cardknox.js

```diff
@@ -17,195 +17,232 @@
+            //onSuccess
+            //perform your own validation here...
+            if (document.getElementsByName("xCardNum")[0].value === "") {
+              $(document).trigger("cardknoxError", "Card Number Required");
+              return false;
+            }
+            if (document.getElementsByName("xCVV")[0].value === "") {
+              $(document).trigger("cardknoxError", "CVV Required");
               return false;
-            },
-            //30 second timeout
-            30000
-          );
-          return false;
-        }
-        return true;
-      },
-  
-      onCardknoxResponse: function () {
-        var xExp = document
-          .getElementById("cardknox-card-expiry")
-          .value.replace(/\s|\//g, "");
-        if (xExp.length != 4) {
-          $(document).trigger("cardknoxError", "Invalid expiration date");
-          return false;
-        }
-        console.log("onCardknoxResponse");
-        wc_cardknox_form.form.append(
-          "<input type='hidden' class='xExp' name='xExp' value='" + xExp + "'/>"
+            }
+            console.log("Success");
+            wc_cardknox_form.onCardknoxResponse();
+          },
+          function () {
+            //onError
+            console.log("error");
+            $(document).trigger(
+              "cardknoxError",
+              document.getElementById("ifieldsError").textContent
+            );
+            return false;
+          },
+          //30 second timeout
+          30000
         );
-        wc_cardknox_form.form.submit();
-      },
-  
-      reset: function () {
-        $("#cardknox-card-cvc, #cardknox-card-number").val("");
-        $(".xExp").remove();
-      },
-  
-      onIfieldloaded: function () {
-        enableLogging();
-        setAccount(wc_cardknox_params.key, "wordpress", "0.1.2");
-        var card_style = {
-          outline: "none",
-          border: "0",
-          "border-left-color": "rgb(67, 69, 75)",
-          padding: "0.6180469716em",
-          width: "225px",
-          height: "auto",
-          "background-color": wc_cardknox_params.bgcolor,
-          "font-weight": "inherit",
-        };
-        var cvv_style = {
-          outline: "none",
-          border: "0",
-          "border-left-color": "rgb(67, 69, 75)",
-          padding: "0.6180469716em",
-          width: "106px",
-          height: "auto",
-          "background-color": wc_cardknox_params.bgcolor,
-          "font-weight": "inherit",
-        };
-        setIfieldStyle("card-number", card_style);
-        setIfieldStyle("cvv", cvv_style);
-      },
-    };
-  
-    wc_cardknox_form.init();
+        return false;
+      }
+      return true;
+    },
+
+    onCardknoxResponse: function () {
+      var xExp = document
+        .getElementById("cardknox-card-expiry")
+        .value.replace(/\s|\//g, "");
+      if (xExp.length != 4) {
+        $(document).trigger(
+          "cardknoxError",
+          "Expiration must be in the future"
```

Comment:
I don't understand the changes you made here. In the previous version you were validating that the exp. is in the future, which is why I suggested the text change. Now it seems we are just validating the length so the old message should be used. Am I correct in my understanding?
```suggestion
          "Invalid expiration date"
```